tests/mongo.js

```diff
@@ -0,0 +1,74 @@
+const request = require("supertest");
+require("./../src/server.js");
+const mongoose = require("mongoose");
+
+const connectionString = "mongodb://admin:example@localhost:27017/test";
+const endpoint = "http://localhost:8000";
+
+const dbComment = {
+  "id": "g84jrs",
+  "created_utc": "2023-03-31T18:01:59.709Z",
+};
+
+const { MongoMemoryServer } = require("mongodb-memory-server");
+let mongod = null;
+
+const connectDB = async () => {
+  try {
+    let dbUrl = connectionString;
+    if (MongoMemoryServer) {
+      mongod = await MongoMemoryServer.create();
+      dbUrl = mongod.getUri();
+    }
+
+    const conn = await mongoose.connect(dbUrl, {
+      useNewUrlParser: true,
+      useUnifiedTopology: true,
+      useFindAndModify: false,
+    });
+
+    console.log(`MongoDB connected: ${conn.connection.host}`);
+  } catch (err) {
+    console.log(err);
+    process.exit(1);
+  }
+};
+
+const disconnectDB = async () => {
+  try {
+    await mongoose.connection.close();
+    if (mongod) {
+      await mongod.stop();
+    }
+  } catch (err) {
+    console.log(err);
+    process.exit(1);
+  }
+};
+
+describe("MongoDB test", () => {
+  beforeAll(() => {
+    connectDB();
+  });
+
+  afterAll(() => {
+    disconnectDB();
+    server.close();
+  });
+
+  describe("Save comment to db", function () {
+    it("POST /api/comments", function (done) {
+      request(endpoint)
+        .post("/api/comments")
+        .send(dbComment)
+        .expect(200)
+        .expect(function (res) {
+          res.body == dbComment;
+        })
+        .catch((error) => {
+          console.log("MongoDB ERROR: ", error);
+        });
+      done();
```

Comment:
Right now you probably are seeing an error along the lines of `Cannot log after tests are done. Did you forget to wait for something async in your test?`

The particular syntax seems to be [promise ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) based - meaning that `expect` and `catch` functions would be likely called after `done` is called. 

There are two ways about it - you could declare function as `async` and use `await` to make execution linear - I believe that was used in the example you've referenced (https://medium.com/weekly-webtips/express-js-testing-mocking-mongodb-46c3797a201), look at the `async` & `await` usage

![image](https://user-images.githubusercontent.com/800419/229290998-a683d05e-66fb-4b07-8be9-2f08ce4966cf.png)


Or you could add `done()` call to the `.finally()` block at the end of the chain.